conda/recipes/versions.yaml

```diff
@@ -133,7 +133,7 @@ scipy_version:
 setuptools_version:
   - '>=49,<50'
 spdlog_version:
-  - '=1.7.0'
+  - '>=1.8.2'
```

Comment:
Should we pin this specifically to `1.8.2` or at least pin to `>=1.8.2,<1.9.0a0` or `>=1.8.2,<2.0.0a0`?

Reply:
Hopefully `spdlog` would be stable enough going forward to not break our builds, but I'd be fine either way.

Reply:
I'm definitely not comfortable allowing it to roll to new major versions, but I'd defer to @harrism @jrhemstad or others for if they have thoughts here on how they want to handle this dependency

Reply:
Spdlog already broke on cuda once where it worked in an older version. I don't think cUDA stability is their priority. So probably should lock it to specific releases unless we contribute tests and CI to spdlog.

Reply:
@kkraus14 looking at their release history, I don't think there are planned major changes between different point releases.

@harrism the break was caused by `fmt`, right? Our CIs would definitely catch any breaking changes, and then we can restrict the versions. Would that be enough?

I'm hesitant to add unnecessary version restrictions as they also affect downstream projects. For example, if someone wants to use a newer a version of `spdlog` and they also depend on rmm/cudf, they would be out of luck if we lock the version.

Reply:
That's true. OK.

Reply:
@rongou I agree with @kkraus14 that we should restrict it to at least `>=1.8.2,<1.9.0a0` or `>=1.8.2,<2.0.0a0`

You've recently reviewed their docs, I'll let you decided which restriction we should use. I understand your comment, but due to the issues we've experienced in the past and the issues with keeping working conda builds over a release we cannot leave this unpinned

Reply:
The same pinning should reflect in `rmm` when the decision is made

Reply:
Added pinning.